web/src/rtl.ts

```diff
@@ -109,42 +123,50 @@ function get_bidi_class(ch) {
  * @param {string} str The string to get its direction.
  * @returns {'ltr' | 'rtl'}
  */
-export function get_direction(str) {
+export function get_direction(str: string): "ltr" | "rtl" {
     let isolations = 0;
+
     for (const ch of str) {
-        const bidi_class = get_bidi_class(ch.codePointAt(0));
+        const bidi_class = get_bidi_class(ch.codePointAt(0) as number);
```

Comment:
@sbansal1999 what should be done in case of `undefined`? Here are the possible solutions:-

1. We change the implementation of `get_bidi_class` function, to account for `ch === undefined`. So, we probably return `Other` for `undefined`? Idk if this is intended because I'm not well versed with the codebase yet.
2. We check for `undefined` in the `for` loop inside `get_direction` function. If `undefined` is found, we skip that iteration of the loop.

Reply:
Practically speaking `ch.codePointAt(0)` will never return `undefined` but TypeScript has no way of knowing it. You will need to handle the case.
I don't think changing the implementation of `get_bidi_class` is a good idea. Instead we can use `||` to handle the `undefined` case.

Reply:
@sbansal1999 Now I get it. So, `.codePointAt(0)` here will never return `undefined` at runtime because we're looping over the characters of a string. If the string is empty, the loop will never get triggered, so `.codePointAt(0)` will never encounter an empty string, and therefore never return `undefined`.

Since, `undefined` doesn't exist at runtime, shouldn't we leave it like I've done? What's the point of handling `undefined` when it can never happen at runtime?

Reply:
When it is *locally* guaranteed that an expression is never `undefined` at runtime, you can use a [non-null assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-).